tests/test_context.py

```diff
@@ -0,0 +1,108 @@
+import strax
+from strax.testutils import Records, run_id
+import tempfile
+import numpy as np
+from hypothesis import given
+import hypothesis.strategies as st
+import typing as ty
+
+
+def _apply_function_to_data(function
+                            ) -> ty.Tuple[np.ndarray, np.ndarray]:
+    """
+    Inner core to test apply function to data
+    :param function: some callable function that takes thee positional
+        arguments
+    :return: records, records with function applied
+    """
+    with tempfile.TemporaryDirectory() as temp_dir:
+        st = strax.Context(storage=strax.DataDirectory(temp_dir,
+                                                       deep_scan=True),
+                           register=[Records])
+
+        # First load normal records
+        records = st.get_array(run_id, 'records')
+
+        # Next update the context and apply
+        st.set_context_config({'apply_data_function': function})
+        changed_records = st.get_array(run_id, 'records')
+    return records, changed_records
+
+
+def test_apply_pass_to_data():
+    """
+    What happens if we apply a function that does nothing (well,
+        nothing hopefully)
+    :return: None
+    """
+
+    def nothing(data, r, t):
+        return data
+
+    r, r_changed = _apply_function_to_data(nothing)
+    assert np.all(r == r_changed)
+
+
+@given(st.integers(min_value=-10, max_value=10))
+def test_apply_ch_shift_to_data(magic_shift: int):
+    """
+    Apply some magic shift number to the channel field and check the results
+    :param magic_shift: some number to check that we can shift the
+        channel field with
+    :return: None
+    """
+
+    def shift_channel(data, r, t):
+        """Add a magic number to the channel field in the data"""
+        res = data.copy()
+        res['channel'] += magic_shift
+        return res
+
+    r, r_changed = _apply_function_to_data(shift_channel)
+    assert len(r) == len(r_changed)
+    assert np.all((r_changed['channel'] - (r['channel'] + magic_shift)) == 0)
+
+
+def test_apply_drop_data():
+    """
+    What if we drop random portions of the data, do we get the right results?
+    :return: None
+    """
+
+    class Drop:
+        """Small class to keep track of the number of dropped fields"""

```

Comment:
```suggestion
        """Small class to keep track of the number of dropped rows"""

```
First I associated fields with the field names of the structured array.